src/System.Reflection/tests/TypeDerivedTests.cs

```diff
@@ -0,0 +1,29 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Reflection.Tests
+{
+    internal class MockDerivedType : MockType
```

Comment:
Nit: A better name for this may be `TypeWithNullUnderlyingSystemType`. It is a very specific mock type; not an arbitrary one.